### PR TITLE
Switch to a static package format design

### PR DESCRIPTION
**Pull request by negz**
### Description of your changes

This PR updates the v2 package metadata design to reflect our decision that packages should be static - i.e. should be files in an OCI image rather than the product of executing an OCI image's entrypoint.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: design/one-pager-package-format-v2.md
Comment (negz) [inline design/one-pager-package-format-v2.md:278]: @rbwsam @rathpc I feel like we could probably take the existing detail about package metadata from the v1 package manager design and the UI schema design and refactor it into something like `design-doc-recommended-additional-metadata.md`. What do you think?

File: design/one-pager-package-format-v2.md
Comment (hasheddan) [inline design/one-pager-package-format-v2.md:65]: nitpick:
```suggestion
format is compiled to a `/package.yaml` file that is then placed in an OCI image.
```

File: design/one-pager-package-format-v2.md
Comment (hasheddan) [inline design/one-pager-package-format-v2.md:70]: Do we have a strict requirement that it be single-layer?

File: design/one-pager-package-format-v2.md
Comment (hasheddan) [inline design/one-pager-package-format-v2.md:276]: nitpick:
```suggestion
package should contain. In the future Crossplane may recommend standard additional
```

File: design/one-pager-package-format-v2.md
Comment (rathpc) [inline design/one-pager-package-format-v2.md:251]:
```suggestion
passthrough metadata and supplementary metadata.
```

File: design/one-pager-package-format-v2.md
Comment (rathpc) [reply design/one-pager-package-format-v2.md:278]: That makes sense to me. I also think some updates need to be made to https://github.com/crossplane/crossplane/blob/master/design/design-doc-upbound-ui-spec.md but that will be independent of that refactoring

**Review by rathpc — CHANGES_REQUESTED** [review]

File: design/one-pager-package-format-v2.md
Comment (lukeweber) [reply design/one-pager-package-format-v2.md:70]: I think package.yaml being the first layer is fine requirement for the future. Images would be well suited maybe in an extra layer, but this is kind of digging into implementation and maybe best to wait until ideas like oras can land with more time in the future to be more opinionated if it makes sense.

File: design/one-pager-package-format-v2.md
Comment (lukeweber) [inline design/one-pager-package-format-v2.md:179]: `Configuration` instead of `config` just for consistency

File: design/one-pager-package-format-v2.md
Comment (negz) [reply design/one-pager-package-format-v2.md:70]: Yeah, I was going more for 'first layer' than 'single layer'. I was mostly intending to communicate 'you don't need to include an entire OS in your package'. Will rephrase.

File: design/one-pager-package-format-v2.md
Comment (negz) [reply design/one-pager-package-format-v2.md:276]: This is correct, albeit British-leaning, grammar.

File: design/one-pager-package-format-v2.md
Comment (negz) [reply design/one-pager-package-format-v2.md:251]: It is important that our metadata be supple.

File: design/one-pager-package-format-v2.md
Comment (negz) [reply design/one-pager-package-format-v2.md:179]: I got lazy because my automatic 80 char line wrapping tool won't wrap comments inside YAML snippets for some reason. Fixed.

**Comment by negz** [comment]
All feedback addressed - PTAL.

**Review by hasheddan — APPROVED** [review]

**Review by lukeweber — APPROVED** [review]

**Review by rbwsam — APPROVED** [review]

**Pull request merged**
